config/config.go

```diff
@@ -25,6 +25,58 @@ import (
 	tracing "github.com/uber/jaeger-client-go/config"
 )
 
+// Config number limitations
+const (
+	MinPortNumber            = 1
+	MaxPortNumber            = 65535
+	MinTokenLimit            = 1
+	MaxTokenLimit            = 2000
+	MinLogSize               = 1024
+	MaxLogSize               = 1 << 32
+	MinLogDays               = 1
+	MaxLogDays               = 2 * 365
+	MinBackups               = 1
+	MaxBackups               = 1000
+	MinSlowLogThreshold      = 100
+	MaxSlowLogThreshold      = 300 * 1000
+	MinExpensiveLogThreshold = 10
+	MaxExpensiveLogThreshold = 1 << 20
+	MinQueryLogLength        = 100
+	MaxQueryLogLength        = 10000
+	MinMetricsInterval       = 1
+	MaxMetricsInterval       = 120
+	MinProcs                 = 0
+	MaxProcs                 = 1000
+	MinRetryLimit            = 0
+	MaxRetryLimit            = 100
+	MinJoinConcurrency       = 1
+	MaxJoinConcurrency       = 100
+	MinStmtCountLimit        = 1000
+	MaxStmtCountLimit        = 1 << 20
+	MinPlanCacheCap          = 1
+	MaxPlanCacheCap          = 10000
```

Comment:
How about 1<<20 ? It can be used as long as our users have large enough memory.

Reply:
For a global level cache capacity, I think 10000 is large enough.